src/main/kotlin/net/weavemc/loader/api/event/EventTypes.kt

```diff
@@ -112,28 +180,74 @@ public class RenderWorldEvent(public val partialTicks: Float) : Event()
 
 public class RenderHandEvent(public val partialTicks: Float) : CancellableEvent()
 
+/**
+ * This event is called when you connect to a server.
+ *
+ * @param ip The IP address of the server.
+ * @param port The port of the server.
+ */
 public class ServerConnectEvent(
     public val ip: String,
     public val port: Int,
 ) : Event() {
+
+    /**
+     * The [Server Data][ServerData] object of the server being connected to.
+     */
     public val serverData: ServerData = Minecraft.getMinecraft().currentServerData
 }
 
+/**
+ * This event is called when the game is starting. Like [Tick Event][TickEvent], it
```

Comment:
Probably don't mention that it's similar to TickEvent, rather do what you did with the other events with sub classes.